kernel/configs/cruel.conf

```diff
@@ -25,3 +28,6 @@ CONFIG_SECURITY_SELINUX_SWITCH=y
 # CONFIG_DEFAULT_HTCP is not set
 # CONFIG_DEFAULT_WESTWOOD is not set
 CONFIG_CC_WERROR=n
+CONFIG_STATE_NOTIFIER=y
+CONFIG_MALI_KUTF=y
```

Comment:
Why do you want to enable CONFIG_MALI_KUTF?

Reply:
From drivers/gpu/arm/b_r12p0/tests/kutf/Kconfig it's testing framework
```
config MALI_KUTF                                                                 
 tristate "Mali Kernel Unit Test Framework"                                      
 default m                                                                       
 help                                                                            
   Enables MALI testing framework. To compile it as a module,                    
   choose M here - this will generate a single module called kutf.
```

Reply:
As the name says, it is part of Mali. It increases performance 5%

Reply:
How could testing framework increase the performance? I looked at the main source file  drivers/gpu/arm/b_r10p0/tests/kutf/kutf_suite.c It only bloats the kernel and add test/debug functions. Even samsung builds it only as module.

Reply:
The testing is worsening the performance a bit, but if there is a problem with the GPU it can alert the user. It only checks at boot anyeay